hatchery/src/instance.rs

```diff
@@ -205,6 +208,12 @@ impl Instance {
         self.id
     }
 
+    pub fn set_snapshot_id(&mut self, snapshot_id: SnapshotId) {
```

Comment:
Is there a reason for this to be public? Seems like internal-only api to me. consider `pub(crate)` if neccesary